### ISSUE TITLE
Issues Compilation S021619

### ISSUE THREAD

**Issue by Team4681** [garykac/amhs-robotics-4681#31]
1. [Issue #32] Hatch needs to extend to grab sucker **(Code should be written but has not been tested)**
2. [Issue #33] In walker back legs and front legs are backwards somehow
3. [Issue #34] Grabber right is not needed 
4. [Issue #35] Why are we inverting the lifter?
5. [Issue #37] Shift x and y buttons so X switches between on and off and y extends and retracts **(Code should be written but has not been tested)**
6. [FIXED] Grabber wheels don't move
7. [Issue #36] Implement Limit Switch code in lowering method on lifter class **(Code should be written but has not been tested)**
8. [FIXED] **Doesnt Need fix** Updated so grabber stops when RT pressed
9. [Issue #37] Laser Does NOT work

**Comment by cmarley3-14** [comment]
** Addressing issue 31.1**
Shouldn't the piston the sucker is attached to be extending to grab the hatch?  
Also, something that we could utilize is `set(!get)`.
For example say that button X is supposed to extend and retract a piston.
The code would look something like this:
``` java
if (m_stick.getRawButtonPressed(kButtonX) {
    m_solenoid.set(!m_solenoid.get());
}
```
We just set the state of the solenoid to the opposite of what it currently is.

**Addressing issue 31.2**
The robot climbs backwards. We'll just have to recode the Walker if something's different.

**Addressing issue 31.3**
Isn't there a motor on the right arm of the grabber we need to operate though?

**31.4**
We may have to invert the lifter because of the way something was installed. Think of it like this.  If you look at a shaft spinning clockwise from the left, then it'll look like it's spinning counter-clockwise from the right side. Maybe setting a motor positive is just spinning it clockwise, from the *left* perspective, so we have to account for something or other or whatever...

**31.5**
See 31.1 for button duality.

**31.6**
Last I saw, they weren't wired, and one of the constants got commented out.

**31.7**
I'll write example code later. (It's 10:30 PM).

**31.9**
Is it still returning values that are semi-accurate at the least?
